### DISCUSSION TITLE
人工审核插件支持通过环境变量动态生成备选项

### DISCUSSION THREAD

**Discussion by finecci** [Tencent/bk-ci#5076] [Q&A]
场景：

1. 假设一个部署的流水线
![image](https://user-images.githubusercontent.com/3276549/131838876-57ae680b-c7b0-4cfb-9482-1da55847107e.png)

2. 该流水线某一步骤获取到最终部署目标列表（因为可用的目标会发生变化，所以这里需要动态获取），设置环境变量DestList 
![image](https://user-images.githubusercontent.com/3276549/131838944-79dd5bdb-e827-4f89-9f5c-f7645b8e38e1.png)

3. 人工审核节点配置选择目标列表
![image](https://user-images.githubusercontent.com/3276549/131839089-2e1f6864-9691-4f30-9497-b3dd1672f933.png)

问题：

目前下拉列表的备选项无法通过传入的变量动态生成，只能写死，无法满足目前的使用场景
![image](https://user-images.githubusercontent.com/3276549/131839409-a553e2a9-66d7-4107-bf78-17c74cdbed78.png)

希望能够支持通过传入参数构造备选项，支持流水线运行时手动选择目标参数的取值，向后传递使用

**Comment by iccengan** [comment]
这个能力还需要进一步评估。
想再咨询下，你们这里期望配置的具体内容是什么？全量列表内容多么？可不可以把全量的列表先配置进去进行选择？

**Reply by finecci** [reply]
> 这个能力还需要进一步评估。 想再咨询下，你们这里期望配置的具体内容是什么？全量列表内容多么？可不可以把全量的列表先配置进去进行选择？

这里设计为分set发布，每次动态获取可发布的set名称，手动选择
前面的流程会获取到这个list，期望能动态生成选项

写死备选项的方式无法同步到set增删变化

**Comment by fayewong-v5** [comment]
第二步已经评估出影响范围，获取到最终部署的目标列表，下游步骤直接使用这个列表部署即可
为什么还需要在人工审核时让审核人去这些目标列表里再次选择将要部署的目标呢？

**Reply by finecci** [reply]
> 第二步已经评估出影响范围，获取到最终部署的目标列表，下游步骤直接使用这个列表部署即可 为什么还需要在人工审核时让审核人去这些目标列表里再次选择将要部署的目标呢？

是需要从这个list中手动选择一个进行发布，不是全部都发
